File: DSA/HR/breaking_the_records.java
import java.util.Arrays;
import java.util.List;

class Result {
    public static List<Integer> breakingRecords(List<Integer> scores) {
        int currMin = scores.get(0), currMax = scores.get(0);
        int minCount = 0, maxCount = 0;

        for (int i = 1; i < scores.size(); i++) {
            int score = scores.get(i);
            if (score > currMax) {
                currMax = score;
                maxCount++;
            }
            if (score < currMin) {
                currMin = score;
                minCount++;
            }
        }

        return Arrays.asList(maxCount, minCount);
    }
}

File: DSA/HR/breaking_the_records.cpp
#include <vector>

using namespace std;

vector<int> breakingRecords(vector<int> scores)
{
    int min{scores[0]}, max{scores[0]};
    int minCount{}, maxCount{};

    for (int score : scores)
    {
        if (score < min)
        {
            minCount++;
            min = score;
        }
        else if (score > max)
        {
            maxCount++;
            max = score;
        }
    }

    return {maxCount, minCount};
}